### PR TITLE
[INFINITY-1365] Prepare refine resources

### PR DESCRIPTION
**Pull request by gabrielhartmann**
This should be sufficient changes such that a one line change in `Capabilities` will enable resource refinement.

I tested this on `testing/master` and `testing/pull/1524` manually.

### CODE REVIEW

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/validation/CapabilityValidator.java [hunk 1]
```diff
@@ -22,14 +19,10 @@ public void validate(ServiceSpec serviceSpec) throws CapabilityValidationExcepti
...
     private void validateGpuPolicy(ServiceSpec serviceSpec) throws CapabilityValidationException {
-        try {
-            if (DefaultService.serviceSpecRequestsGpuResources(serviceSpec)
-                    && !Capabilities.getInstance().supportsGpuResource()) {
-                throw new CapabilityValidationException(
-                        "This cluster's DC/OS version does not support setting GPU_RESOURCE");
-            }
-        } catch (IOException e) {
-            throw new CapabilityValidationException("Failed to determine capabilities: " + e.getMessage());
+        if (DefaultService.serviceSpecRequestsGpuResources(serviceSpec)
```
Comment (gabrielhartmann) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/validation/CapabilityValidator.java]: The diff and the one below look like git is just not diffing well.  This should be just removing the try/catch.  Can another set of eyes validate, please.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceBuilder.java [hunk 2]
```diff
@@ -190,8 +190,9 @@ public ResourceBuilder setMesosResource(MesosResource mesosResource) {
     public Resource build() {
         Resource.Builder builder =
                 mesosResource == null ? Resource.newBuilder() : mesosResource.getResource().toBuilder();
-        builder.setName(resourceName).setRole(preReservedRole);
-        builder.setType(value.getType());
+        builder.setName(resourceName)
```
Comment (gabrielhartmann) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceBuilder.java]: Changes here and below are the bug fix, I needed to make to resource refinement work again on `testing/pull/1524`

**Comment by gabrielhartmann** [comment]
test this please

**Comment by gabrielhartmann** [comment]
retest this please

File: sdk/scheduler/src/test/java/com/mesosphere/sdk/testutils/DefaultCapabilitiesTestSuite.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * This class provides default a context with the default set of capabilities.
+ */
+public abstract class DefaultCapabilitiesTestSuite {
```
Comment (ArtRand) [inline sdk/scheduler/src/test/java/com/mesosphere/sdk/testutils/DefaultCapabilitiesTestSuite.java]: 👍

File: sdk/scheduler/src/test/java/com/mesosphere/sdk/testutils/DefaultCapabilitiesTestSuite.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import com.mesosphere.sdk.dcos.Capabilities;
+import com.mesosphere.sdk.dcos.ResourceRefinementCapabilityContext;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * This class provides default a context with the default set of capabilities.
```
Comment (ArtRand) [inline sdk/scheduler/src/test/java/com/mesosphere/sdk/testutils/DefaultCapabilitiesTestSuite.java]: 👍

File: sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/ResourceBuilderTest.java [hunk 5]
```diff
@@ -85,6 +86,51 @@ public void testNewFromResourceSpecRefined() {
...
+                    value,
+                    TestConstants.ROLE,
+                    TestConstants.PRE_RESERVED_ROLE,
+                    TestConstants.PRINCIPAL,
+                    "CPUS_ENV_KEY");
+            ResourceBuilder resourceBuilder = ResourceBuilder.fromSpec(resourceSpec, Optional.empty());
+
+            Protos.Resource resource = resourceBuilder.build();
+            Assert.assertEquals(2, resource.getReservationsCount());
+            validateScalarResourceRefined(TestConstants.PRE_RESERVED_ROLE, resource);
```
Comment (mrbrowning) [inline sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/ResourceBuilderTest.java]: I don't think helper functions like this are valuable in tests. For the most part I want to be able to read a single test case and know what conditions are under test, `validateScalarResourceRefined` doesn't do that and I have to jump around to understand it. If un-DRYing it is onerous I think it's a sign that further decomposition is warranted.

Comment (gabrielhartmann) [inline sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/ResourceBuilderTest.java]: Actually change to this helper was unneeded, thanks for mentioning it.  I think I understand your point, but will have to think about this.  The point of the helper is to validate that a certain format is maintained across many tests.  I guess you're saying a single test should validate this, and then things should be structured that we need not worry that in some other situation something else could occur, as code structure would make such a scenario impossible.  I'll think about it.

**Comment by gabrielhartmann** [comment]
retest this please

**Pull request merged**